src/hooks/use-products.jsx

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react';
+import { getProducts, getCategorizedProducts } from './../data/API';
+import { useParams } from 'react-router-dom';
+
+export default function useProducts(sort) {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
```

Comment:
```suggestion
  const [error, setError] = useState("");
```
로직상 error 에는 문자열이 담길 예정이므로 빈 문자열로 초기화 했습니다.
아무것도 넣지 않아 undefined 로 state 가 초기화 되면
앞으로 여기에 문자열이 들어갈지, 객체가 들어갈지 알수가 없습니다.
그러면 해당 state가 무슨 역할을 할지 추측을 할 수 없어서 가독성이 매우 안좋아요